### PR TITLE
[Perf] Intergrate Tile Kernels `head_compute_mix_kernel` for Deepseek-V4

### PR DESCRIPTION
**Pull request by Isotr0py**
## Purpose
- Related issue: https://github.com/vllm-project/vllm/issues/40902
- Port https://github.com/deepseek-ai/TileKernels/blob/main/tile_kernels/mhc/head_compute_mix_kernel.py for deepseek-v4

## Test Plan
Test on 4xGB200
```
vllm serve deepseek-ai/DeepSeek-V4-Flash -tp 4 -ep --trust-remote-code --max-num-seq 256 --kv-cache-dtype fp8
```

## Test Result
Tested with `SPEED-Bench`:
```
Generating 500 prompts from SPEED-Bench (throughput_16k/low_entropy, truncated to 10240 tokens)...
Oversampled SPEED-Bench from 302 to 500 total samples.
SPEED-Bench categories: low_entropy:302
Generated 500 prompts in 4.42s
Warming up with 50 requests...
Warmup run completed.
```

### This PR (`51fdf186b`)
```
================= Serving Benchmark Result =================
Successful requests:                     500       
Failed requests:                         0         
Benchmark duration (s):                  100.18    
Total input tokens:                      5120000   
Total generated tokens:                  116183    
Request throughput (req/s):              4.99      
Output token throughput (tok/s):         1159.71   
Peak output token throughput (tok/s):    6663.00   
Peak concurrent requests:                500.00    
Total token throughput (tok/s):          52266.24  
--------------------Time to First Token---------------------
Mean TTFT (ms):                          48130.70  
Median TTFT (ms):                        47242.82  
P99 TTFT (ms):                           93541.79  
----------Time per Output Token (excl. 1st token)-----------
Mean TPOT (ms):                          146.48    
Median TPOT (ms):                        180.33    
P99 TPOT (ms):                           202.04    
--------------------Inter-token Latency---------------------
Mean ITL (ms):                           143.60    
Median ITL (ms):                         174.37    
P99 ITL (ms):                            253.67    
============================================================
```

### Main branch (`a269744e9`)
```
================= Serving Benchmark Result =================
Successful requests:                     500       
Failed requests:                         0         
Benchmark duration (s):                  109.42    
Total input tokens:                      5120000   
Total generated tokens:                  118505    
Request throughput (req/s):              4.57      
Output token throughput (tok/s):         1083.05   
Peak output token throughput (tok/s):    6842.00   
Peak concurrent requests:                500.00    
Total token throughput (tok/s):          47876.24  
--------------------Time to First Token---------------------
Mean TTFT (ms):                          51223.21  
Median TTFT (ms):                        48162.24  
P99 TTFT (ms):                           102489.60 
----------Time per Output Token (excl. 1st token)-----------
Mean TPOT (ms):                          165.20    
Median TPOT (ms):                        204.16    
P99 TPOT (ms):                           218.93    
--------------------Inter-token Latency---------------------
Mean ITL (ms):                           162.68    
Median ITL (ms):                         200.46    
P99 ITL (ms):                            263.33    
============================================================
```

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

### CODE REVIEW

File: vllm/model_executor/models/deepseek_v4.py [hunk 1]
```diff
@@ -1451,14 +1451,28 @@ def hc_head(
...
-    mixes = F.linear(x, hc_fn) * rsqrt
-    pre = torch.sigmoid(mixes * hc_scale + hc_base) + hc_eps
-    y = torch.sum(pre.unsqueeze(-1) * x.view(shape), dim=1)
-    return y.to(dtype)
+    hc_mult = hidden_states.shape[-2]
+    hidden_size = hidden_states.shape[-1]
+    outer_shape = hidden_states.shape[:-2]
+    hs_flat = hidden_states.view(-1, hc_mult, hidden_size)
+    num_tokens = hs_flat.shape[0]
+    h_block = math.gcd(512, hidden_size)
```
Comment (zyongye) [inline vllm/model_executor/models/deepseek_v4.py]: Yea where is 512 coming from. Can we tune it better?

Comment (Isotr0py) [reply vllm/model_executor/models/deepseek_v4.py]: Actually, it's same as `h_blk` in `mhc_post_tilelang`:
https://github.com/vllm-project/vllm/blob/a07642667db1284ad2128d7f9ef089e6b0d24a4c/vllm/model_executor/layers/mhc.py#L359-L369

I compared both `h_block=1024, n_thr=128` and `h_block=512, n_thr=64`, the performance is not different too much. So I keep it aligned with `mhc_post_tilelang` now.

**Review by zyongye — APPROVED** [review]
Local test on DEP8 B300 V4-PRO. Accuracy verified with GPQA score 90.
